computer-vision/robot_tracking.cpp

```diff
@@ -0,0 +1,167 @@
+#include <opencv2/highgui.hpp>
+#include <opencv2/aruco.hpp>
+#include <opencv2/calib3d.hpp>
+
+#include <iostream>
+
+#include "protobuf/protocol.pb.h"
+#include "protobuf/computer_vision.pb.h"
+#include "error/error.h"
+#include "logging.h"
+#include "udp_protocol.h"
+
+#define ARUCO_CENTER 42
+#define ARUCO_WEATHERVANE 17
+
+struct RobotTrackingSettings {
+    int image_width;
+    int image_height;
+    cv::Mat camera_matrix;
+    cv::Mat distortion_coefficients;
+
+    void load(const std::string& file)
+    {
+        int flags = cv::FileStorage::READ; // cv::FileStorage::FORMAT_YAML
+        cv::FileStorage fs = cv::FileStorage(file, flags);
+
+        cv::read(fs["image_width"], image_width, -1);
+        cv::read(fs["image_height"], image_height, -1);
+        cv::read(fs["camera_matrix"], camera_matrix);
+        cv::read(fs["distortion_coefficients"], distortion_coefficients);
+    }
+};
+
+enum WeathiervaneDirection {
+    NORTH = 0,
+    SOUTH = 1
+};
+
+WeathiervaneDirection weathervane_direction(cv::Vec3d rvec, cv::Vec3d tvec)
+{
+    cv::Mat R;
+    cv::Rodrigues(rvec, R);
```

Comment:
aaaaah Rodrigues convention :volcano:

Reply:
What is rodrigues convention ?

Reply:
no, just thesis flashbacks #vietnam

Reply:
the day we lost @antoinealb https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula